### ISSUE TITLE
Crash when start (Android)

### ISSUE THREAD

**Issue by KarlosQ** [rebeccahughes/react-native-device-info#152]
The app crash me when start, I try with manual installation and with react-native link.

Does not display error message.

react-native-device-info: 0.99
RN: 0.41.2
Android I test (4.4, 7.1)

**Comment by danprat92** [comment]
Had the same issue and this thread helped me https://github.com/rebeccahughes/react-native-device-info/issues/128.

**Comment by stephanedemotte** [comment]
same here, confict with react-native-maps

**Comment by KarlosQ** [comment]
@stephanedemotte Yes, but i don't know how to fix.

**Comment by danprat92** [comment]
@stephanedemotte try opening node_modules/react-native-maps/android/build.gradle and change to

`compile "com.google.android.gms:play-services-maps:+"`

**Comment by damusnet** [comment]
Had the same problem, here is how I fixed it. Basically you have a Google Play Services version mismatch. `react-native-maps` for example is set fix on 9.8.0. From `./node_modules/react-native-maps/android/build.gradle`:

```
dependencies {
  compile "com.facebook.react:react-native:+"
  compile "com.google.android.gms:play-services-base:9.8.0"
  compile "com.google.android.gms:play-services-maps:9.8.0"
}
```

Meanwhile, `react-native-device-info` has:

```
dependencies {
    compile 'com.facebook.react:react-native:+'
    compile 'com.google.android.gms:play-services-gcm:+'
}
```

Here is my `./android/app/build.gradle`:

```
dependencies {
    compile(project(':react-native-device-info')){
        exclude group: 'com.google.android.gms', module: 'play-services-gcm'
    }
    compile 'com.google.android.gms:play-services-gcm:9.8.0'
    compile project(':react-native-maps')
    compile fileTree(dir: "libs", include: ["*.jar"])
    compile "com.android.support:appcompat-v7:23.0.1"
    compile "com.facebook.react:react-native:+"  // From node_modules
}
```

Also, I made sure which Google Play Services version is actually installed on my Emulator by going to Settings >> App >> Google Play Services (it was 9.8.79 for me).

So basically, look at all your dependencies, if they have Google Play Services requirements. Determine the lowest common denominator, and which version is on your emulator. Exclude Google Play Services from the compilation of your dependencies that are on an unspecified or otherwise problematic version. Specify by hand which version you really want for the Google Play Service in question.

**Comment by patlux** [comment]
@damusnet: After I added the line to my `./android/app/build.gradle` every time I run `react-native link` the link command adds a second `compile project(':react-native-device-info') ` (<-- without the brackets) before the manually added line.

```
dependencies {
    compile project(':react-native-device-info')
    compile(project(':react-native-device-info')){
        exclude group: 'com.google.android.gms', module: 'play-services-gcm'
    }
    // ....
    compile fileTree(dir: "libs", include: ["*.jar"])
    compile "com.android.support:appcompat-v7:23.0.1"
    compile "com.facebook.react:react-native:+"  // From node_modules
}
```

Because the pattern looks for the brackets:
https://github.com/facebook/react-native/blob/master/local-cli/link/android/patches/makeBuildPatch.js

How to avoid this?

**Comment by damusnet** [comment]
Unfortunately I honestly don't know. Maybe you could bring it up with the
`react-native link` maintainers, they would probably be interested in this
feedback. This tool is more of a helper and you should always double check
what it does before committing anyway.

Le 12 avr. 2017 6:26 AM, "Patrick Wozniak" <notifications@github.com> a
écrit :

**Comment by ramavadhmaurya** [comment]
Change your app/build.gradle dependencies to look like this:

dependencies {
...
compile(project(':react-native-device-info')) {
exclude group: 'com.google.android.gms'
}
}

and then go to cd android
clean your gradlew file
and come back to your root directory and then run your application.
It is working for me :)

BR,
Ram Avadh Maurya

**Comment by babarbahadur** [comment]
I was also having the same issue and i simply commented these 3 lines in node_modules/react-native-device-info/android/build.gradle.. It works for me!!!
**Note: I wasn't using firebase, I was only using react-native-maps & geolocation-location library. Please make sure you have manually linked the library**

if(firebaseIidVersion){
**// implementation "com.google.firebase:firebase-iid:$firebaseIidVersion"**
}else{
**// def iidVersion = safeExtGet('googlePlayServicesIidVersion',** safeExtGet('googlePlayServicesVersion', '17.0.0'))
**// implementation "com.google.android.gms:play-services-iid:$iidVersion"**
}
